tls/s2n_quic_support.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/s2n_quic_support.h"
+
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls13.h"
+#include "utils/s2n_safety.h"
+
+int s2n_connection_enable_quic(struct s2n_connection *conn)
+{
+    /* The QUIC RFC is not yet finalized, so all QUIC APIs are
+     * considered experimental and subject to change.
+     * They should only be used for testing purposes.
+     */
+    ENSURE_POSIX(S2N_IN_TEST, S2N_ERR_NOT_IN_TEST);
```

Comment:
I'm not a huge fan of conflating test mode with enabling QUIC support, as test mode could be opting in to other behavior outside of QUIC. Could it be an explicit define instead (e.g. `S2N_UNSTABLE_ENABLE_QUIC`)?

Reply:
My concern at the moment is just that this is an totally incomplete feature that should only be called from our tests for now.

I could just remove the S2N_API flags for now instead.

Reply:
We talked offline: we'll just remove this check since you have to pull in the internal header anyway. The S2N_API attribute will stay, though, so it can be consumed.